widget/list.go

```diff
@@ -761,6 +761,10 @@ func (l *listLayout) updateList(newOnly bool) {
 		for _, vis := range l.visible {
 			l.setupListItem(vis.item, vis.id, l.list.focused && l.list.currentFocus == vis.id)
 		}
+
+		// a full refresh means drain the pool of excess old items as well
```

Comment:
Can we elaborate on the why here - that the old items may have an old theme applied? I can imagine this piece of code, and the current comment, becoming cryptic with time since it's not obvious why we'd be draining the pool. Also would be useful if we revisit this some time later with smarter refreshing to avoid having to do this.

Reply:
When we call Refresh it is iterating visible items to refresh them.
But the items in the cache are being left in old theme.
Is the sentence there just needing more "why" in it?

Reply:
yeah I think the code comment needs the info about the old theme - so someone coming to this code later on after this PR is long forgotten knows why it was added

Reply:
OK thanks, I hope the update helps